shell-ui/src/initFederation/ConfigurationProviders.tsx

```diff
@@ -318,9 +341,52 @@ export function useDiscoveredViews(): ViewDefinition[] {
 
     return [];
   }) as ViewDefinition[];
-
   return discoveredViews;
 }
+
+type FederatedRoute = {
+  app: SolutionUI;
+  view: View;
+};
+export function useFederatedRoutes(): FederatedRoute[] {
+  const { retrieveConfiguration } = useConfigRetriever();
+  const { retrieveDeployedApps } = useDeployedAppsRetriever();
+  const deployedApps = retrieveDeployedApps();
+
+  deployedApps.forEach((app) => {
+    if (app.appHistoryBasePath.endsWith('/')) {
+      throw new Error(
+        `appHistoryBasePath of app ${app.name} ends with a /, this is not allowed`,
+      );
+    }
```

Comment:
Do we really want to run this on each render of the component holding this hook ? I think we can at least move a good part of this hook content into a useMemo